### PR TITLE
feat: add Transaction filtering info

### PR DESCRIPTION
**Pull request by dimazhornyk**
# Description

Add the Transaction Filtering documentation

## Additional context

Add a new block to the transaction lifecycle doc that explains the possible options to filter the transactions coming from either L1 contracts or API.

### CODE REVIEW

File: content/10.zk-stack/05.concepts/00.transaction-lifecycle.md [hunk 1]
```diff
@@ -159,3 +159,36 @@ These fields are handled by our SDKs.
 
 Each of these transaction types ensures that while ZKsync Era remains closely aligned with Ethereum standards,
 it also optimizes for its Layer 2 specific needs and functionalities.
+
+---
+### Transaction filtering
+The ZK Chain operators can both the transactions initiated from L1 and the ones submitted via API.
```
Comment (githubdoramon) [inline content/10.zk-stack/05.concepts/00.transaction-lifecycle.md:165]: Missing a "filter" in the sentence maybe @dimazhornyk ?

**Review by githubdoramon — CHANGES_REQUESTED** [review]
Wondering if we shouls add these 2 as components on https://docs.zksync.io/zk-stack/components

Maybe a high level menu as "Tx Filtering", and then the 2 components mentioned here. Thoughts @dimazhornyk ?

File: content/10.zk-stack/05.concepts/00.transaction-lifecycle.md (hunk 1 above)
Comment (dimazhornyk) [reply content/10.zk-stack/05.concepts/00.transaction-lifecycle.md:165]: Yes, I will add it. And yeah, it makes sense to have it separately, it does belong there more than in the transaction lifecycle

**Review by githubdoramon — DISMISSED** [review]

**Review by itsacoyote — CHANGES_REQUESTED** [review]
Preferably we want to have yaml frontmatter at the top of the page to have more control over metadata.  
For this example:

```md
---
title: Transaction filtering
description: 
---
``` 

You'll need to remove the `#Transaction filtering` as the title replaces the H1.

The description field is a way to create a short description that fits in social link shares. Otherwise it will use the first sentence in the page which usually looks awkward. Leaving it blank will just leave the description in the share link blank if you don't want to add anything description-wise.

An example of the social link (this one is for /build/connect-to-zksync): 
Screenshot 2024-06-17 at 1 13 28 PM

File: content/10.zk-stack/10.components/110.transaction-filtering.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
+title: Transaction filtering
+description: Learn how to filter transactions on the ZK chain, including L1→L2 and L2 transactions.
+---
+
+The ZK Chain operators can filter both the transactions initiated from L1 and the ones submitted via API directly to L2.
+
+### L1->L2 Transactions
```
Comment (itsacoyote) [inline content/10.zk-stack/10.components/110.transaction-filtering.md:8]: Sorry I missed these in the initial review, can you update all the H3 headers to H2?

**Review by itsacoyote — APPROVED** [review]

**Pull request merged**